### ISSUE TITLE
Implement intersphinx entries web viewer (devtools?)

### ISSUE THREAD

**Issue by webknjaz** [sphinx-doc/sphinx#8211]
**Is your feature request related to a problem? Please describe.**

I'm often frustrated when I need to reference things from other sphinx sites. While I know about `python -m sphinx.ext.intersphinx https://docs.python.org/3/objects.inv`, it is not very user-friendly. It tries to represent a lot of columns with data making the console output hard to follow.

For example, it's easy to find some name with `less` or `grep` but then it's also easy to miss the role name because it appears only once before a bunch of entries. This is known to create some extra confusion (https://github.com/sphinx-doc/sphinx/issues/6659).

**Describe the solution you'd like**

I think a better alternative for a CLI command w/o args would be a web-based view.

It's pretty easy to represent it as an HTML table. Also, it'd be possible to add some extra filtering, search, and even usage example.

I see two options of what could be done:
1) Add a built-in web-page into the docs that exposes everything present on the current website (just another representation for `objects.inv`). This is useful for others trying to link against this site.
2) Another option is to have some way of serving a web-interface on the client-side (a project that links against others). This would basically have multiple intersphinx targets (as specified in `conf.py`) shown as a web UI.

**Describe alternatives you've considered**

I've already mentioned this elsewhere (https://github.com/sphinx-doc/sphinx/issues/6659#issuecomment-687567225) — I've made a simple site based on GitHub Actions workflow just for myself that takes a bunch of intersphinx links and generates static HTML pages. It is very simple but it should be a good demo of why this is useful — https://webknjaz.github.io/intersphinx-untangled.

(src: https://github.com/webknjaz/intersphinx-untangled)

**Comment by tk0miya** [comment]
It sounds good to me. But I think no resource to develop such a feature to sphinx core for our team. So it would be better to develop it as a 3rd party tool or extension.

**Comment by webknjaz** [comment]
Interesting. I'm only concerned that intersphihx's APIs are private

**Comment by tk0miya** [comment]
What do you mean "intersphinx's APIs"? I think the `sphinx.util.inventroy` module helps you.

**Comment by webknjaz** [comment]
@tk0miya it's about the option 2: https://github.com/webknjaz/intersphinx-untangled/blob/b798e9d/.github/workflows/build-gh-pages.yml#L36

**Comment by tk0miya** [comment]
`fetch_inventory` is not an API. It's application code... No plan to make it public.

**Comment by webknjaz** [comment]
That is what I'm talking about. Such things can also be called private APIs. That is also the reason I suggested this feature in core.

**Comment by tk0miya** [comment]
Essentially, it downloads an inventory file and parse it by `InventoryFile` class. I don't think it is not common feature. I'd not like to make it public. And I don't think it does not prevent making 3rd party tools.

**Comment by webknjaz** [comment]
Yes, I know. But it promotes duplicating the logic elsewhere. I just think that a viewer should be an integral part of intersphinx itself.

P.S. You said earlier that the core team won't have time for this but would you consider an external contribution if somebody were to implement this and send a PR?

**Comment by tk0miya** [comment]
>P.S. You said earlier that the core team won't have time for this but would you consider an external contribution if somebody were to implement this and send a PR?

We also don't have time for maintenance, not only creating a new module. So I'm okay if somebody will become a maintainer of the module. How about joining our team?

Note: I can't say it will be bundled with the Sphinx package. It might be better to provide it as a separated (but an official) package.

**Comment by webknjaz** [comment]
> How about joining our team?

I have too much on my plate right now but if I end up sending a PR, I'll be maintaining this feature.

**Comment by astrojuanlu** [comment]
For the record, https://sphobjinv.readthedocs.io/ exists:

```
$ sphobjinv suggest doc/build/html/objects.inv as_rst -st 50
  Name                                                     Score
--------------------------------------------------------  -------
:py:method:`sphobjinv.data.SuperDataObj.as_rst`             60
:std:doc:`cli/implementation/parser`                        57
:py:module:`sphobjinv.cli.parser`                           50
:py:method:`sphobjinv.data.SuperDataObj.as_str`             50
:py:method:`sphobjinv.inventory.Inventory.objects_rst`      50
```

**Comment by webknjaz** [comment]
Yes, but that only works if you know what you're looking for. Sometimes you don't.

**Comment by webknjaz** [comment]
xref https://github.com/orgs/sphinx-doc/discussions/12152
